### DISCUSSION TITLE
agent count always shows zero

### DISCUSSION THREAD

**Discussion by daorlov** [amidaware/tacticalrmm#1778] [Q&A]
Good day everyone, in the upper right corner it always shows zero, although agents are active, and the icons opposite them are green
![123](https://github.com/amidaware/tacticalrmm/assets/161603504/168b90d5-a10d-4384-ab0d-758812e4328d)

**Comment by dinger1986** [comment]
how did you install?

**Reply by daorlov** [reply]
https://docs.tacticalrmm.com/install_server/

**Reply by dinger1986** [reply]
theres multiple ways, on that help me help you, give me more info

**Reply by wh1te909** [reply]
what's the output of `sudo systemctl status --full --no-pager daphne`

**Reply by Avessalom608** [reply]
root@ubuntu:~# sudo systemctl status --full --no-pager daphne
● daphne.service - uvicorn daemon v1
     Loaded: loaded (/etc/systemd/system/daphne.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2024-02-26 08:25:21 UTC; 2 days ago
   Main PID: 27169 (uvicorn)
      Tasks: 7 (limit: 4534)
     Memory: 102.2M
        CPU: 3min 7.503s
     CGroup: /system.slice/daphne.service
             └─27169 /rmm/api/env/bin/python3.11 /rmm/api/env/bin/uvicorn --uds /rmm/daphne.sock "--forwarded-allow-ips=*" tacticalrmm.asgi:application

Feb 29 07:09:11 ubuntu uvicorn[27169]: INFO:     connection open
Feb 29 07:15:35 ubuntu uvicorn[27169]: INFO:     connection closed
Feb 29 07:24:57 ubuntu uvicorn[27169]: INFO:     ('172.16.50.26', 0) - "WebSocket /ws/dashinfo/?access_token=REDACTED" [accepted]
Feb 29 07:24:57 ubuntu uvicorn[27169]: INFO:     connection open
Feb 29 07:25:04 ubuntu uvicorn[27169]: INFO:     connection closed
Feb 29 07:25:10 ubuntu uvicorn[27169]: INFO:     ('172.16.50.26', 0) - "WebSocket /ws/dashinfo/?access_token=REDACTED" [accepted]
Feb 29 07:25:10 ubuntu uvicorn[27169]: INFO:     connection open
Feb 29 07:31:50 ubuntu uvicorn[27169]: INFO:     connection closed
Feb 29 07:31:57 ubuntu uvicorn[27169]: INFO:     ('172.16.50.26', 0) - "WebSocket /ws/dashinfo/?access_token=REDACTED" [accepted]
Feb 29 07:31:57 ubuntu uvicorn[27169]: INFO:     connection open

**Reply by silversword411** [reply]
You don't manage your server from user `root`.

**Reply by Avessalom608** [reply]
@silversword411 I'm sorry

tactical@ubuntu:~$ systemctl status --full --no-pager daphne
● daphne.service - uvicorn daemon v1
     Loaded: loaded (/etc/systemd/system/daphne.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2024-03-05 10:23:23 UTC; 24h ago
   Main PID: 710 (uvicorn)
      Tasks: 7 (limit: 4534)
     Memory: 118.7M
        CPU: 1min 841ms
     CGroup: /system.slice/daphne.service
             └─710 /rmm/api/env/bin/python3.11 /rmm/api/env/bin/uvicorn --uds /rmm/daphne.sock "--forwarded-allow-ips=*" tacticalrmm.asgi:application

Mar 06 11:12:48 ubuntu uvicorn[710]:     await super().send({"type": "websocket.accept", "subprotocol": subprotocol})
Mar 06 11:12:48 ubuntu uvicorn[710]:   File "/rmm/api/env/lib/python3.11/site-packages/channels/consumer.py", line 81, in send
Mar 06 11:12:48 ubuntu uvicorn[710]:     await self.base_send(message)
Mar 06 11:12:48 ubuntu uvicorn[710]:   File "/rmm/api/env/lib/python3.11/site-packages/channels/sessions.py", line 226, in send
Mar 06 11:12:48 ubuntu uvicorn[710]:     return await self.real_send(message)
Mar 06 11:12:48 ubuntu uvicorn[710]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Mar 06 11:12:48 ubuntu uvicorn[710]:   File "/rmm/api/env/lib/python3.11/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 379, in asgi_send
Mar 06 11:12:48 ubuntu uvicorn[710]:     raise RuntimeError(msg % message_type)
Mar 06 11:12:48 ubuntu uvicorn[710]: RuntimeError: Expected ASGI message 'websocket.http.response.body' but got 'websocket.accept'.
Mar 06 11:12:48 ubuntu uvicorn[710]: INFO:     connection closed

**Reply by Avessalom608** [reply]
@dinger1986 
The installation was traditional, without any changes. Option 1: Easy installation on VPS. In protected mode

**Reply by dinger1986** [reply]
what does protected mode mean?

**Reply by Avessalom608** [reply]
@dinger1986 
> what does protected mode mean?

Run install.sh without the key --insecure

**Reply by silversword411** [reply]
Here's what I'd recommend.
Reboot
From the correct user, run your ./update.sh --force
To fix what I'd guess is probably broken permissions on the server at a minimum.


If it's still broken, open a ticket with support 
https://docs.tacticalrmm.com/support/#commercial-support

and setup ssh access (lower in the page) and we'll login and have a look.

Include the sequence of events from when it was working, to when it broke.

**Reply by Avessalom608** [reply]
Unfortunately, running the script did not produce results, but thanks for the answer

**Comment by lqdmist** [comment]
Any solution? I just updated to 0.19.1 in the hope this would be fixed but no.

**Reply by dinger1986** [reply]
did you use --insecure when you installed?

**Reply by lqdmist** [reply]
No but the issue was already present before updating, I thought maybe the update would fix this.

**Reply by wh1te909** [reply]
this is not a bug with the software, there's no issues with agent counts showing 0. something specific to your install or system.

**Reply by silversword411** [reply]
Do you have another proxy/load balancer in front of the server processing traffic on 443?

You haven't given any technical details on your install other than didn't use `--insecure`
https://docs.tacticalrmm.com/faq/#if-you-received-this-link

**Reply by lqdmist** [reply]
The agent count was showing the correct value before. The (Debian 12) VM experienced a sudden crash resulting in a read-only file system (the cause of this is still unknown). Rebooting this VM brought the application back up and running except for the agent count.
The application is installed natively (no docker), so only the default Nginx is listening on 443.
We ran the troubleshooting script which didn't notice any issues

**Comment by Trapulo** [comment]
I have same problem running 1.0.0. If I full refresh (F5) the page, numbers are ok

**Reply by silversword411** [reply]
You probably have an incorrectly configured 2nd proxy.

**Reply by Trapulo** [reply]
What do you mean? All works in the admin panel, but this counter sometime is zero until a reload.

**Reply by silversword411** [reply]
I mean your unsupported 2nd proxy is misconfigured, and they're probably slicing sessions (UDP based websocket ones) that should be sitting open waiting for data to flow thru the session created when you hit F5.
Your custom proxy or other network appliance is ignoring requested TTL session values and/or artificially closing them and then doesn't let that traffic thru once it's closed the session in the NAT or proxy states table.
